src/Application/Commands/DeleteStalePaymentDetails.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Commands;
+
+use Stripe\Service\ChargeService;
+use Stripe\Service\CustomerService;
+use Stripe\Service\PaymentMethodService;
+use Stripe\StripeClient;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Deletes payment methods from Stripe if:
+ *   1. they (or more specifically, their Customer) are 24h+ old; and
+ *   2. they do not belong to a Person with a password set after mid-April 2023; and
+ *   3. they are a credit/debit card based method; and
+ *   4. they have not been used for any complete donations.
```

Comment:
what's the reason for condition 4?

Also is there any way a person account could be more than 24 hours old but still have a valid JWT if it's been renewed a few times? Not sure if it's possible to renew the JWT without a password.

Reply:
Right. I think we should be deleting even if they were used for completed (or paid) donations, but it makes sense to test it out first.

Reply:
On point 2 - we want need to keep payment methods around if they belong to a person with a password set in the past, don't we? Not only keep them for people who set passwords after mid-april 2023. I know it's harder from matchbot to detect which accounts had passwords set in the past.

Reply:
@bdsl I think those methods are essential to keep only when (4) is also true, so I believe we're pretty much covered with these conditions. This is what I was getting at re. only cards which have been used exclusively for _failed_ donations being at risk of deletion when we wouldn't ideally have it – which IMO is not a big deal.

Reply:
Right, I agree that it's not to be a problem to detach payment methods that were never used for a successful donation. But I do think eventually we should be detaching all payment methods from all old passwordless accounts. (and also probably removing them from id DB as afaik they have no purpose there, but that's another question)

So yes these four conditions together I think are good for now, we just need to know that we can't get rid of #4 without also changing #2.